app/cdap/components/RecipeList/ViewAllRecipes.tsx

```diff
@@ -40,6 +44,12 @@ const ViewAllRecipies = () => {
 
   return (
     <>
+      {showRecipesDrawer && (
+        <RecipeDetails
```

Comment:
1. RecipeDetails should not contain drawer.   
2. Should be a generic drawer container and container content should be loaded based on the selected context menu option. For Example, when view is selected , RecipeDetails  is shown . If Edit is clicked, EditRecipe component should be loaded inside the drawer component.
3. This was we can load any content inside the  Drawer Container without defining Drawer structure in each component.
4. Could use DrawerWidget component here, I think which was developed by Divami team only. (https://github.com/cdapio/cdap-ui/blob/feature/wrangler-2022/app/cdap/components/common/DrawerWidget/index.tsx) 
Let me know if you need more details on this, I can help you.

Reply:
Done